controllers/pod_node_module_reconciler.go

```diff
@@ -70,8 +72,21 @@ func (pnmr *PodNodeModuleReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	if !podutils.IsPodReady(&pod) || !pod.DeletionTimestamp.IsZero() {
 		logger.Info("Unlabeling node")
 
-		if err := pnmr.deleteLabel(ctx, nodeName, labelName); err != nil {
-			return ctrl.Result{}, fmt.Errorf("could not unlabel node %s: %v", nodeName, err)
+		// Make sure we don't already have a new running pod before unlabeling the node
+		labelSelector := client.MatchingLabels{constants.ModuleNameLabel: moduleName}
+		fieldSelector := client.MatchingFields{"spec.nodeName": nodeName}
+		var modulePodsList v1.PodList
+		err := pnmr.client.List(ctx, &modulePodsList, labelSelector, fieldSelector)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get list of all pods for module %s on node %s: %v", moduleName, nodeName, err)
+		}
+		if len(modulePodsList.Items) == 0 {
```

Comment:
We should also delete if there are non-ready pods or pods with deletionTimestamp set. Will change it.

Reply:
Fixed.